src/Microsoft.DotNet.Tools.Publish/Program.cs

```diff
@@ -34,13 +35,16 @@ public static int Main(string[] args)
 
             app.OnExecute(() =>
             {
-                if (!CheckArg(framework))
+                if (framework.HasValue() && IsUnsupportedFramework(framework.Value()))
```

Comment:
can you move the framework.HasValue() inside the IsUnsupported? Would make this conditional here cleaner.